File: apps/Applications/ios/Local For Vocal/Local For Vocal/Sources/Views/Pages/BasketPageView.swift
import SwiftUI

struct BasketPageView: View {
    @EnvironmentObject var basketManager: BasketManager
    @Environment(\.presentationMode) var presentationMode
    @State private var isSearching = false
    var groceryTab: Binding<GroceryTab>?  // Optional: when inside GroceryContainerView
    @State private var isCheckoutActive = false

    var body: some View {
        NavigationView {
            VStack(spacing: 0) {
                // Header
                HStack(spacing: 12) {
                    Button(action: {
                        if let tab = groceryTab {
                            tab.wrappedValue = .grocery
                        } else {
                            presentationMode.wrappedValue.dismiss()
                        }
                    }) {
                        Image(systemName: "arrow.left")
                            .font(.system(size: 18, weight: .medium))
                            .foregroundColor(Color(hex: "#111827"))
                    }

                    Text("My Basket (\(basketManager.basketCount) Items)")
                        .font(.system(size: 18, weight: .medium))
                        .foregroundColor(Color(hex: "#111827"))

                    Spacer()

                    Button(action: {
                        isSearching = true
                    }) {
                        Image(systemName: "magnifyingglass")
                            .font(.system(size: 18, weight: .medium))
                            .foregroundColor(Color(hex: "#111827"))
                    }
                }
                .padding()
                .background(Color.white)
                .shadow(color: Color.black.opacity(0.05), radius: 2, x: 0, y: 2)
                .zIndex(1)

                if basketManager.items.isEmpty {
                    BasketEmptyStateView()
                } else {
                    ScrollView {
                        VStack(spacing: 0) {
                            // 1. Delivery Banner
                            DeliveryBannerView()  // Reuse from CartPageView if compatible, or redefine
                                .padding(.horizontal, 16)
                                .padding(.top, 16)

                            // 2. Items List
                            VStack(spacing: 0) {  // Divider styling
                                ForEach(basketManager.items) { item in
                                    BasketItemCell(item: item)
                                        .padding(.vertical, 16)
                                        .overlay(
                                            Divider(),
                                            alignment: .bottom
                                        )
                                }
                            }
                            .padding(.horizontal, 16)
                            .background(Color.white)
                            .padding(.bottom, 16)

                            // 3. Bill Details
                            BillDetailsView(total: basketManager.basketTotal)  // Reuse
                                .padding(.horizontal, 16)
                                .padding(.bottom, 16)

                            // 4. Savings Banner
                            if basketManager.basketSavings > 0 {
                                SavingsBannerView(savings: basketManager.basketSavings)
                                    .padding(.horizontal, 16)
                                    .padding(.bottom, 100)
                            } else {
                                Color.clear.frame(height: 100)
                            }
                        }
                    }
                    .background(Color(hex: "#F3F4F6"))
                }

                // Checkout Bar (if not empty)
                if !basketManager.items.isEmpty {
                    VStack(spacing: 0) {
                        Divider()
                        HStack {
                            VStack(alignment: .leading) {
                                Text("Total")
                                    .font(.system(size: 12))
                                    .foregroundColor(Color(hex: "#6B7280"))
                                Text("₹\(Int(basketManager.basketTotal + 2))")  // +2 handling
                                    .font(.system(size: 18, weight: .bold))
                                    .foregroundColor(Color(hex: "#111827"))
                            }
                            Spacer()
                            Button(action: {
                                isCheckoutActive = true
                            }) {
                                HStack {
                                    Text("Proceed to Pay")
                                        .font(.system(size: 14, weight: .bold))
                                    Image(systemName: "arrow.right")
                                }
                                .foregroundColor(.white)
                                .padding(.vertical, 12)
                                .padding(.horizontal, 24)
                                .background(Color(hex: "#15803d"))
                                .cornerRadius(8)
                            }
                        }
                        .padding(16)
                        .background(Color.white)
                    }
                    .zIndex(2)
                }
            }
            .background(Color(hex: "#F3F4F6").edgesIgnoringSafeArea(.all))
            .background(
                NavigationLink(
                    destination: CheckoutView(
                        items: basketManager.items.map { item in
                            CheckoutViewModel.CheckoutItem(
                                product: item.product,
                                quantity: item.quantity,
                                selectedOfferIds: []
                            )
                        }
                    ),
                    isActive: $isCheckoutActive
                ) { EmptyView() }
            )
            .navigationBarHidden(true)
        }  // NavigationView
        .navigationViewStyle(.stack)
        .fullScreenCover(isPresented: $isSearching) {
            GroceryGlobalSearchView()
        }
    }
}

// MARK: - Subviews

struct BasketEmptyStateView: View {
    var body: some View {
        ScrollView {
            VStack(spacing: 0) {
                VStack(alignment: .center, spacing: 20) {
                    if let url = URL(
                        string: "https://cdn-icons-png.flaticon.com/512/11329/11329060.png")
                    {
                        AsyncImage(url: url) { image in
                            image.resizable().aspectRatio(contentMode: .fit)
                        } placeholder: {
                            Color.gray.opacity(0.1)
                        }
                        .frame(width: 150, height: 150)
                    }

                    Text("Your basket is empty!")
                        .font(.system(size: 18, weight: .bold))
                        .foregroundColor(Color(hex: "#111827"))

                    Button(action: {
                        // Shop now action
                    }) {
                        Text("Shop now")
                            .font(.system(size: 14, weight: .semibold))
                            .foregroundColor(.white)
                            .padding(.vertical, 12)
                            .padding(.horizontal, 48)
                            .background(Color(hex: "#2563EB"))
                            .cornerRadius(4)
                    }
                }
                .padding(.bottom, 30)
                .padding(.top, 10)
                .frame(maxWidth: .infinity)
                .background(Color.white)

                // Recommendations Section Placeholder
                // Need to implement Steal Deal / Try it Buy it logic here if required
            }
        }
        .background(Color(hex: "#F3F4F6"))
    }
}

struct BasketItemCell: View {
    let item: CartItem
    @EnvironmentObject var basketManager: BasketManager

    var body: some View {
        HStack(alignment: .top, spacing: 12) {
            // Image
            ZStack {
                Color(hex: "#F9FAFB")
                if let firstImage = item.product.images.first {
                    let urlString =
                        firstImage.hasPrefix("http")
                        ? firstImage : "\(APIService.shared.imageHost)\(firstImage)"
                    AsyncImage(url: URL(string: urlString)) { image in
                        image.resizable().aspectRatio(contentMode: .fit)
                    } placeholder: {
                        Color.gray.opacity(0.1)
                    }
                }
            }
            .frame(width: 60, height: 60)
            .cornerRadius(8)
            .overlay(
                RoundedRectangle(cornerRadius: 8)
                    .stroke(Color(hex: "#E5E7EB"), lineWidth: 1)
            )

            // Info
            VStack(alignment: .leading, spacing: 4) {
                Text(item.product.name)
                    .font(.system(size: 14, weight: .medium))
                    .foregroundColor(Color(hex: "#1F2937"))
                    .lineLimit(2)

                Text(item.product.subtitle ?? "1 pc")  // Weight from description/subtitle if mapped
                    .font(.system(size: 12))
                    .foregroundColor(Color(hex: "#6B7280"))

                HStack(alignment: .firstTextBaseline, spacing: 8) {
                    Text("₹\(Int(item.product.price))")
                        .font(.system(size: 14, weight: .bold))
                        .foregroundColor(Color(hex: "#111827"))

                    if let mrp = item.product.mrp, mrp > item.product.price {
                        Text("₹\(Int(mrp))")
                            .font(.system(size: 12))
                            .foregroundColor(Color(hex: "#9CA3AF"))
                            .strikethrough()
                    }
                }
            }

            Spacer()

            // Quantity Control
            HStack(spacing: 0) {
                Button(action: {
                    basketManager.updateQuantity(
                        productId: item.productId, quantity: item.quantity - 1)
                }) {
                    Text("-")
                        .font(.system(size: 16, weight: .bold))
                        .foregroundColor(.white)
                        .frame(width: 28, height: 32)
                }

                Text("\(item.quantity)")
                    .font(.system(size: 14, weight: .semibold))
                    .foregroundColor(.white)
                    .frame(minWidth: 20)

                Button(action: {
                    basketManager.updateQuantity(
                        productId: item.productId, quantity: item.quantity + 1)
                }) {
                    Text("+")
                        .font(.system(size: 16, weight: .bold))
                        .foregroundColor(.white)
                        .frame(width: 28, height: 32)
                }
            }

            .background(Color(hex: "#15803d"))
            .cornerRadius(6)
        }
    }
}

struct DeliveryBannerView: View {
    var body: some View {
        HStack {
            Image(systemName: "location.fill")
                .foregroundColor(Color(hex: "#15803d"))
            Text("Delivery to ")
                .font(.system(size: 14))
                .foregroundColor(Color(hex: "#374151"))
                + Text("Home - 560066")
                .font(.system(size: 14, weight: .bold))
                .foregroundColor(Color(hex: "#111827"))
            Spacer()
            Button("Change") {
                // Action
            }
            .font(.system(size: 12, weight: .bold))
            .foregroundColor(Color(hex: "#15803d"))
        }
        .padding(12)
        .background(Color(hex: "#dcfce7"))
        .cornerRadius(8)
        .overlay(
            RoundedRectangle(cornerRadius: 8)
                .stroke(Color(hex: "#86efac"), lineWidth: 1)
        )
    }
}

struct BillDetailsView: View {
    let total: Double
    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            Text("Bill Details")
                .font(.system(size: 14, weight: .bold))
                .foregroundColor(Color(hex: "#374151"))

            HStack {
                Text("Item Total")
                    .font(.system(size: 12))
                    .foregroundColor(Color(hex: "#4b5563"))
                Spacer()
                Text("₹\(Int(total))")
            }
            HStack {
                Text("Delivery Fee")
                    .font(.system(size: 12))
                    .foregroundColor(Color(hex: "#4b5563"))
                Spacer()
                Text("₹25")
                    .strikethrough()
                    .foregroundColor(Color(hex: "#9ca3af"))
                    + Text(" Free")
                    .foregroundColor(Color(hex: "#15803d"))
            }
            HStack {
                Text("Handling Charge")
                    .font(.system(size: 12))
                    .foregroundColor(Color(hex: "#4b5563"))
                Spacer()
                Text("₹2")
            }
            Divider()
            HStack {
                Text("To Pay")
                    .font(.system(size: 14, weight: .bold))
                    .foregroundColor(Color(hex: "#111827"))
                Spacer()
                Text("₹\(Int(total + 2))")
                    .font(.system(size: 14, weight: .bold))
                    .foregroundColor(Color(hex: "#111827"))
            }
        }
        .padding(12)
        .background(Color.white)
        .cornerRadius(8)
    }
}

struct SavingsBannerView: View {
    let savings: Double

    var body: some View {
        HStack {
            Image(systemName: "tag.fill")
                .foregroundColor(Color(hex: "#15803d"))
            Text("You saved ₹\(Int(savings)) on this order")
                .font(.system(size: 12, weight: .semibold))
                .foregroundColor(Color(hex: "#14532d"))
            Spacer()
        }
        .padding(10)
        .background(Color(hex: "#f0fdf4"))
        .cornerRadius(8)
        .overlay(
            RoundedRectangle(cornerRadius: 8)
                .stroke(Color(hex: "#bbf7d0"), style: StrokeStyle(lineWidth: 1, dash: [4]))
        )
    }
}

File: apps/Applications/android/app/src/main/java/com/localforvocalstartup/app/ui/pages/BasketPageView.kt
package com.localforvocalstartup.app.ui.pages

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowBack
import androidx.compose.material.icons.filled.LocationOn
import androidx.compose.material.icons.filled.Search
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextDecoration
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import coil.compose.AsyncImage
import com.localforvocalstartup.app.data.manager.BasketManager
import com.localforvocalstartup.app.data.manager.CartItem
import com.localforvocalstartup.app.ui.viewmodel.CheckoutItem
import com.localforvocalstartup.app.ui.viewmodel.CheckoutViewModel
import androidx.compose.ui.zIndex

@Composable
fun BasketPageView(onDismiss: () -> Unit) {
    // Only items is a StateFlow; count/total/savings are plain computed properties
    val basketItems by BasketManager.items.collectAsState()
    // Derive scalar values reactively by reading from the singleton after items recompose
    val basketCount = basketItems.size
    val basketTotal = BasketManager.basketTotal
    val basketSavings = BasketManager.basketSavings

    var showGlobalSearch by remember { mutableStateOf(false) }
    var showCheckoutFlow by remember { mutableStateOf(false) }

    Box(modifier = Modifier.fillMaxSize().background(Color(0xFFF3F4F6))) {
        Column(modifier = Modifier.fillMaxSize().statusBarsPadding()) {
            
            // Header
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .background(Color.White)
                    .shadow(elevation = 2.dp, shape = RoundedCornerShape(0.dp))
                    .padding(horizontal = 16.dp, vertical = 12.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Icon(
                    imageVector = Icons.Default.ArrowBack, 
                    contentDescription = "Back",
                    tint = Color(0xFF111827),
                    modifier = Modifier.size(24.dp).clickable { onDismiss() }
                )
                Spacer(modifier = Modifier.width(12.dp))
                Text(
                    text = "My Basket ($basketCount Items)",
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Medium,
                    color = Color(0xFF111827)
                )
                Spacer(modifier = Modifier.weight(1f))
                Icon(
                    imageVector = Icons.Default.Search, 
                    contentDescription = "Search",
                    tint = Color(0xFF111827),
                    modifier = Modifier.size(24.dp).clickable { showGlobalSearch = true }
                )
            }

            if (basketItems.isEmpty()) {
                BasketEmptyStateView()
            } else {
                LazyColumn(
                    modifier = Modifier.weight(1f),
                    contentPadding = PaddingValues(bottom = 100.dp) // Leave space for bottom bar
                ) {
                    item {
                        Box(modifier = Modifier.padding(horizontal = 16.dp, vertical = 16.dp)) {
                            DeliveryBannerView()
                        }
                    }

                    item {
                        Column(modifier = Modifier.background(Color.White).padding(bottom = 16.dp)) {
                            basketItems.forEachIndexed { index, item ->
                                BasketItemCell(item)
                                if (index < basketItems.size - 1) {
                                    Divider(color = Color(0xFFE5E7EB))
                                }
                            }
                        }
                    }

                    item {
                        Box(modifier = Modifier.padding(16.dp)) {
                            BillDetailsView(total = basketTotal)
                        }
                    }

                    if (basketSavings > 0) {
                        item {
                            Box(modifier = Modifier.padding(horizontal = 16.dp)) {
                                SavingsBannerView(savings = basketSavings)
                            }
                        }
                    }
                }

                // Checkout Bar
                Column(
                    modifier = Modifier
                        .fillMaxWidth()
                        .background(Color.White)
                        .padding(16.dp)
                        .navigationBarsPadding()
                ) {
                    Divider(color = Color(0xFFE5E7EB), modifier = Modifier.padding(bottom = 16.dp))
                    Row(
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Column {
                            Text("Total", fontSize = 12.sp, color = Color(0xFF6B7280))
                            Text("₹${(basketTotal + 2).toInt()}", fontSize = 18.sp, fontWeight = FontWeight.Bold, color = Color(0xFF111827))
                        }
                        Spacer(modifier = Modifier.weight(1f))
                        Row(
                            modifier = Modifier
                                .background(Color(0xFF15803D), RoundedCornerShape(8.dp))
                                .clickable { showCheckoutFlow = true }
                                .padding(horizontal = 24.dp, vertical = 12.dp),
                            verticalAlignment = Alignment.CenterVertically,
                            horizontalArrangement = Arrangement.spacedBy(8.dp)
                        ) {
                            Text("Proceed to Pay", fontSize = 14.sp, fontWeight = FontWeight.Bold, color = Color.White)
                            // Missing arrow.right equivalent for specific material sizing, using simple unicode or similar if desired.
                        }
                    }
                }
            }
        }
        
        if (showCheckoutFlow) {
            val checkoutItems = basketItems.map { item ->
                CheckoutItem(
                    product = item.product,
                    quantity = item.quantity,
                    selectedOfferIds = emptyList() // Fetch if needed or leave default empty
                )
            }
            val checkoutViewModel = remember { CheckoutViewModel(checkoutItems) }
            
            Box(modifier = Modifier.fillMaxSize().zIndex(10f)) {
                com.localforvocalstartup.app.ui.pages.CheckoutView(
                    viewModel = checkoutViewModel,
                    onBack = { showCheckoutFlow = false }
                )
            }
        }

        if (showGlobalSearch) {
            com.localforvocalstartup.app.ui.pages.GlobalSearchView(onDismiss = { showGlobalSearch = false })
        }
    }
}

@Composable
fun BasketEmptyStateView() {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .background(Color.White)
            .padding(vertical = 30.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(20.dp)
    ) {
        AsyncImage(
            model = "https://cdn-icons-png.flaticon.com/512/11329/11329060.png",
            contentDescription = "Empty Basket",
            contentScale = ContentScale.Fit,
            modifier = Modifier.size(150.dp)
        )
        Text("Your basket is empty!", fontSize = 18.sp, fontWeight = FontWeight.Bold, color = Color(0xFF111827))
        Box(
            modifier = Modifier
                .background(Color(0xFF2563EB), RoundedCornerShape(4.dp))
                .clickable { /* Shop Now */ }
                .padding(horizontal = 48.dp, vertical = 12.dp)
        ) {
            Text("Shop now", fontSize = 14.sp, fontWeight = FontWeight.SemiBold, color = Color.White)
        }
    }
}

@Composable
fun BasketItemCell(item: CartItem) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 16.dp, horizontal = 16.dp),
        horizontalArrangement = Arrangement.spacedBy(12.dp),
        verticalAlignment = Alignment.Top
    ) {
        // Image
        Box(
            modifier = Modifier
                .size(60.dp)
                .background(Color(0xFFF9FAFB), RoundedCornerShape(8.dp))
                .clip(RoundedCornerShape(8.dp)),
            contentAlignment = Alignment.Center
        ) {
            val firstImage = item.product.images?.firstOrNull()
            if (firstImage != null) {
                val urlString = if (firstImage.startsWith("http")) firstImage else "https://api.lfvs.in$firstImage"
                AsyncImage(
                    model = urlString,
                    contentDescription = null,
                    contentScale = ContentScale.Fit,
                    modifier = Modifier.fillMaxSize().padding(4.dp)
                )
            }
        }

        // Details
        Column(
            modifier = Modifier.weight(1f),
            verticalArrangement = Arrangement.spacedBy(4.dp)
        ) {
            Text(
                text = item.product.name ?: "",
                fontSize = 14.sp,
                fontWeight = FontWeight.Medium,
                color = Color(0xFF1F2937),
                maxLines = 2
            )
            Text(
                text = item.product.subtitle ?: "1 pc",
                fontSize = 12.sp,
                color = Color(0xFF6B7280)
            )
            Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                Text("₹${item.product.price.toInt()}", fontSize = 14.sp, fontWeight = FontWeight.Bold, color = Color(0xFF111827))
                val mrp = item.product.mrp ?: 0.0
                if (mrp > item.product.price) {
                    Text(
                        "₹${mrp.toInt()}",
                        fontSize = 12.sp,
                        color = Color(0xFF9CA3AF),
                        textDecoration = TextDecoration.LineThrough
                    )
                }
            }
        }

        // Stepper
        Row(
            modifier = Modifier
                .background(Color(0xFF15803D), RoundedCornerShape(6.dp)),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Box(
                modifier = Modifier.size(28.dp, 32.dp).clickable { BasketManager.updateQuantity(item.productId, item.quantity - 1) },
                contentAlignment = Alignment.Center
            ) {
                Text("-", fontSize = 16.sp, fontWeight = FontWeight.Bold, color = Color.White)
            }
            Text("${item.quantity}", fontSize = 14.sp, fontWeight = FontWeight.SemiBold, color = Color.White)
            Box(
                modifier = Modifier.size(28.dp, 32.dp).clickable { BasketManager.updateQuantity(item.productId, item.quantity + 1) },
                contentAlignment = Alignment.Center
            ) {
                Text("+", fontSize = 16.sp, fontWeight = FontWeight.Bold, color = Color.White)
            }
        }
    }
}

@Composable
fun DeliveryBannerView() {
    val address = "Home - 418" // Placeholder, integrate LocationManager later
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .background(Color(0xFFDCFCE7), RoundedCornerShape(8.dp))
            .padding(12.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Icon(Icons.Default.LocationOn, contentDescription = null, tint = Color(0xFF15803D), modifier = Modifier.size(16.dp))
        Spacer(modifier = Modifier.width(6.dp))
        Text("Delivery to ", fontSize = 14.sp, color = Color(0xFF374151))
        Text(address, fontSize = 14.sp, fontWeight = FontWeight.Bold, color = Color(0xFF111827), maxLines = 1, modifier = Modifier.weight(1f))
        Text("Change", fontSize = 12.sp, fontWeight = FontWeight.Bold, color = Color(0xFF15803D), modifier = Modifier.clickable { /* logic */ })
    }
}

@Composable
fun BillDetailsView(total: Double) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .background(Color.White, RoundedCornerShape(8.dp))
            .padding(12.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        Text("Bill Details", fontSize = 14.sp, fontWeight = FontWeight.Bold, color = Color(0xFF374151))
        
        Row {
            Text("Item Total", fontSize = 12.sp, color = Color(0xFF4B5563))
            Spacer(modifier = Modifier.weight(1f))
            Text("₹${total.toInt()}", fontSize = 12.sp, color = Color(0xFF111827))
        }
        Row {
            Text("Delivery Fee", fontSize = 12.sp, color = Color(0xFF4B5563))
            Spacer(modifier = Modifier.weight(1f))
            Row {
                Text("₹25", fontSize = 12.sp, color = Color(0xFF9CA3AF), textDecoration = TextDecoration.LineThrough)
                Text(" Free", fontSize = 12.sp, color = Color(0xFF15803D))
            }
        }
        Row {
            Text("Handling Charge", fontSize = 12.sp, color = Color(0xFF4B5563))
            Spacer(modifier = Modifier.weight(1f))
            Text("₹2", fontSize = 12.sp, color = Color(0xFF111827))
        }
        Divider(color = Color(0xFFE5E7EB))
        Row {
            Text("To Pay", fontSize = 14.sp, fontWeight = FontWeight.Bold, color = Color(0xFF111827))
            Spacer(modifier = Modifier.weight(1f))
            Text("₹${(total + 2).toInt()}", fontSize = 14.sp, fontWeight = FontWeight.Bold, color = Color(0xFF111827))
        }
    }
}

@Composable
fun SavingsBannerView(savings: Double) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .background(Color(0xFFF0FDF4), RoundedCornerShape(8.dp))
            .padding(10.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        // Tag icon using location or similar, skipping specific tag icon here if missing
        Text("You saved ₹${savings.toInt()} on this order", fontSize = 12.sp, fontWeight = FontWeight.SemiBold, color = Color(0xFF14532D))
    }
}
